collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -94,19 +90,11 @@ public void onClick(View v) {
         addAnswerView(answerLayout);
 
         customTabHelper = new CustomTabHelper();
-        customTabHelper.bindCustomTabsService((AppCompatActivity) context, null);
     }
 
-    private void openUrl() {
-        if (customTabHelper.getPackageName(getContext()).size() != 0) {
-            CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder().build();
-            customTabsIntent.intent.setPackage(customTabHelper.getPackageName(getContext()).get(0));
-            customTabsIntent.launchUrl(getContext(), uri);
-        } else {
-            Intent intent = new Intent(Intent.ACTION_VIEW);
-            intent.setData(uri);
-            getContext().startActivity(intent);
-        }
+    private boolean isUrlEmpty(TextView stringAnswer) {
+        return stringAnswer == null || stringAnswer.getText() == null
+                || "".equals(stringAnswer.getText().toString());
```

Comment:
Any particular reason for the backwards comparison here?

Reply:
I would also prefer this to be `stringAnswer.getText().toString().equals("")`. http://blue-walrus.com/2010/11/reverse-equals-in-java-a-dangerous-technique/

Reply:
I think `stringAnswer.getText().isEmpty()' can also be used

Reply:
I'm not sure that method exists but maybe [TextUtils.isEmpty](https://developer.android.com/reference/android/text/TextUtils.html#isEmpty%28java.lang.CharSequence%29)?

Reply:
https://developer.android.com/reference/java/lang/String.html#isEmpty()